src/strategy/mod.rs

```diff
@@ -0,0 +1,78 @@
+//! Update and reboot strategies.
+
+#![allow(unused)]
+
+use crate::config::inputs;
+use crate::identity::Identity;
+use failure::{bail, Fallible};
+use futures::prelude::*;
+use log::error;
+use serde::Serialize;
+
+mod immediate;
+pub(crate) use immediate::StratImmediate;
+
+#[derive(Clone, Debug, Serialize)]
+pub(crate) enum UpStrategy {
+    Immediate(StratImmediate),
```

Comment:
Minor/optional: I'd just spell those out to be clear (i.e. `UpdateStrategy` and `StrategyImmediate`).

Reply:
Ack, fixed, thanks!
I'm unfortunately often too terse, so please **do** keep highlighting when things get cryptic.